### PR TITLE
CB-13641: (windows) allow for transparent splash screen background co…

### PR DESCRIPTION
**Pull request by IndieSW**
…lor.

### Platforms affected
windows

### What does this PR do?
allows for configuration of transparent splash screen background color.

### What testing has been done on this change?
built app with forked/updated repository and manually tested splash screen background color configuration with RGB and 'transparent' values.

### Checklist
- [x ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ x] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: cordova-js-src/splashscreen.js [hunk 1]
```diff
@@ -69,18 +69,21 @@ function readPreferencesFromCfg(cfg, manifest) {
         splashImageSrc = schema + ':///' + manifest.getSplashScreenImagePath().replace(/\\/g, '/');
 
         bgColor = cfg.getPreferenceValue('SplashScreenBackgroundColor') || bgColor;
-        bgColor = bgColor.replace('0x', '#').replace('0X', '#');
-        if (bgColor.length > 7) {
-            // Remove aplha
-            bgColor = bgColor.slice(0, 1) + bgColor.slice(3, bgColor.length);
-        }
+        bgColor = bgColor.toLowerCase().replace('0x', '#');
```
Comment (janpio) [inline cordova-js-src/splashscreen.js]: This unrelated change doesn't add the possibility of other bugs?

File: cordova-js-src/splashscreen.js [hunk 2]
```diff
@@ -69,18 +69,21 @@ function readPreferencesFromCfg(cfg, manifest) {
...
+
+        if (bgColor !== 'transparent') {
+            if (bgColor.length > 7) {
+                // Remove aplha
```
Comment (janpio) [inline cordova-js-src/splashscreen.js]: maybe fix that typo while you're at it ;)

File: cordova-js-src/splashscreen.js [hunk 3]
```diff
@@ -212,7 +215,7 @@ function exitFullScreen() {
 // Make title bg color match splashscreen bg color
 function colorizeTitleBar() {
     var appView = Windows.UI.ViewManagement.ApplicationView.getForCurrentView();
-    if (isWin10UWP) {
+    if (isWin10UWP && (typeof titleBgColor !== 'undefined')) {
```
Comment (janpio) [inline cordova-js-src/splashscreen.js]: no nicer way to do this but then to check for undefined?
took some time to understand what is going on here, especially with the `titleInitialBgColor ` that depends on this one

File: cordova-js-src/splashscreen.js (hunk 1 above)
Comment (IndieSW) [reply cordova-js-src/splashscreen.js]: I had to force the case of bgColor to compare it to 'transparent' so I combined it with the replacement of '0x' and '0X'. bgColor is used as CSS color value which I believe is case-insensitive.

File: cordova-js-src/splashscreen.js (hunk 2 above)
Comment (IndieSW) [reply cordova-js-src/splashscreen.js]: :)

File: cordova-js-src/splashscreen.js (hunk 3 above)
Comment (IndieSW) [reply cordova-js-src/splashscreen.js]: That's a reasonable point. Both `titleBgColor` and `titleInitialBgColor` are set in one place and used in one place. Adding a variable to track whether or not `bgColor` is transparent seemed like too much. Thoughts?

Comment (janpio) [reply cordova-js-src/splashscreen.js]: First I thought you could just check if bgColor == 'transparent`, but that doesn't exist here. `isTitleBgTransparent` would be a solution and probably more readable, also makes the special case explicit.

**Comment by IndieSW** [comment]
Closing for update.

**Comment by janpio** [comment]
Hm? You could have just committed the changes to your `IndieSW:master` branch for them to show up here automatically.
